src/main/resources/com/tresys/magvar/xsd/magvar.dfdl.xsd

```diff
@@ -50,28 +50,37 @@ SOFTWARE.
         emptyValueDelimiterPolicy="both" representation="text" textNumberRep="standard"
         lengthKind="implicit" encoding="ASCII" encodingErrorPolicy="error" outputNewLine="%CR;%LF;" 
         textNumberJustification="right" textNumberPadCharacter="%SP;" textTrimKind="padChar" textPadKind="padChar"
-        textNumberPattern="###0.0##;-###0.0##" textStandardZeroRep="" />
+        textStandardZeroRep="" />
+      <dfdl:defineFormat name="floatFmt">
+        <dfdl:format textNumberPattern="###0.0##;-###0.0##"/>
+      </dfdl:defineFormat>
     </xs:appinfo>

```

Comment:
Daffodil was working without this added floatFmt. It was unparsing the xs:int elements below without using the textNumberPattern. Seemingly ignoring it for the xs:int elements. 

This is bug https://issues.apache.org/jira/browse/DAFFODIL-2158